NuRadioMC/EvtGen/generator.py

```diff
@@ -408,80 +408,129 @@ def J(E):
 
 def set_volume_attributes(volume, proposal, attributes):
     """
-    helper function that interprets the volume settings and sets the relevant quantities as hdf5 attributes.
+    Helper function that interprets the volume settings and sets the relevant quantities as attributes dictinary.
+    The relevant quantities to define the volume in which neutrinos are generated are (rmin, rmax, zmin, zmax) or
+    (xmin, xmax, ymin, ymax, zmin, zmax) based on the user settings (see explanation blow).
+
+    The user can specify a "fiducial" (mandatory) and "full" (optional) volume. In the end only events are
+    considered (= passed to radio simulation) which produce a shower (over a configurable energy threshold)
+    within the fiducial volume. However, neutrino vertices might be generated outside this fiducial volume
+    but the generated charge lepton (mu or tau) reaches the fiducial volume and produces a trigger
+    (this obviously only makes sense when running proposal). Hence, a "full" volume can be defined in which
+    neutrinos interactions are generated. If the "full" volume is not specified in the case you run proposal,
+    the volume is determind using the energy dependent mean-free-path length of the tau.
+
+    TL;DR
+    - The fiducial volume needs to be chosen large enough such that no events outside of it will trigger.
+    - The full volume is optional and most of the times it is not necessary to specify it (because it is
+    automatically determined).
 
     Parameters
     ----------
-    volume: dictionarty
-        dict specifying the volume
+
+    volume: dict
+        A dictionary specifying the simulation volume. Can be either a cylinder (specified with min/max radius and z-coordinate)
+        or a cube (specified with min/max x-, y-, z-coordinates).
+
+        Cylinder:
+
+        * fiducial_{rmin,rmax,zmin,zmax}: float
+            lower r / upper r / lower z / upper z coordinate of fiducial volume
+
+        * full_{rmin,rmax,zmin,zmax}: float (optional)
+            lower r / upper r / lower z / upper z coordinate of simulated volume
+
+        Cube:
+
+        * fiducial_{xmin,xmax,ymin,ymax,zmin,zmax}: float
+            lower / upper x-, y-, z-coordinate of fiducial volume
+
+        * full_{xmin,xmax,ymin,ymax,zmin,zmax}: float (optional)
+            lower / upper x-, y-, z-coordinate of simulated volume
+
+        Optinal you can also define the horizontal center of the volume (if you want to displace it from the origin)
+
+        * x0, y0: float
+            x-, y-coordinate this shift the center of the simulation volume
+
     proposal: bool
         specifies if secondary interaction via proposal are calculated
+
```

Comment:
Just some nitpicking - parameters in the `Parameters` section of a docstring should not be separated by newlines, they are already clearly separated by the indentation. See [here](https://nu-radio.github.io/NuRadioMC/Introduction/pages/contributing.html#writing-docstrings) for an example. I think it's to some extent okay if you don't want to follow this convention in some of your new code, but I would prefer if you don't 'correct' this in existing classes and functions : )

Reply:
Oh, I was convinced that adding newlines would enforce the coding convention but I was wrong. I have to say I find a parameter section without this separation really hard to read but this might be just me. I will contain myself, sorry.

Reply:
I removed the newlines.